electrumsv/subscription.py

```diff
@@ -166,16 +166,21 @@ def _remove_subscription(self, entry: SubscriptionEntry, owner: SubscriptionOwne
         key = entry.key
         if key.value_type in BYTE_SUBSCRIPTION_TYPES:
             assert type(key.value) is bytes, "subscribed script hashes must be bytes"
-        subscription_id = self._subscription_ids[key]
-        subscriptions = self._subscriptions[key]
-        subscriptions.remove(owner)
-        self._owner_subscriptions[owner].remove(key)
-        del self._owner_subscription_context[(owner, key)]
-        if len(subscriptions) == 0:
-            del self._subscription_ids[key]
-            del self._subscriptions[key]
-            return subscription_id
-        return None
+        try:
+            subscription_id = self._subscription_ids[key]
+            subscriptions = self._subscriptions[key]
+            subscriptions.remove(owner)
+            self._owner_subscriptions[owner].remove(key)
+            del self._owner_subscription_context[(owner, key)]
+            if len(subscriptions) == 0:
+                del self._subscription_ids[key]
+                del self._subscriptions[key]
+                return subscription_id
+            return None
+        except KeyError:
```

Comment:
This `KeyError` suggests that this whole function is unreliable with key errors happening everywhere for random reasons.